### PR TITLE
TRUNK-4988: Convenience method to allow change the configurations more  easily

### PR DESCRIPTION
**Pull request by kdaud**
## Description of what I changed

Convenience method to allow change the configurations more  easily

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-4988

## Checklist: I completed these to help reviewers :)

- [ x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/context/Context.java [hunk 1]
```diff
@@ -576,6 +578,8 @@ public static ProgramWorkflowService getProgramWorkflowService() {
 	 */
 	public static MessageService getMessageService() {
 		MessageService ms = getServiceContext().getMessageService();
+		AbstractApplicationContext context = new ClassPathXmlApplicationContext("applicationContext-service.xml");
+		ms = (MessageService) context.getBean("messageServiceBean", MessageService.class);
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/api/context/Context.java:582]: Can you walk me through what these two lines do and how they differ from the existing line 580?

File: api/src/main/resources/applicationContext-service.xml [hunk 2]
```diff
@@ -597,6 +597,14 @@
 
 	<!--  ******************************  MESSAGE SERVICES  ****************************** -->
 
+    <bean id="messageServiceBean" class="org.openmrs.notification.MessageService">
```
Comment (ibacher) [inline api/src/main/resources/applicationContext-service.xml:600]: How is this bean different from the `messageService` defined below and why do we need it?

**Review by ibacher — CHANGES_REQUESTED** [review]
@kdaud So the ticket mentions some TODO comments which don't seem to be addressed by this PR. It's unfortunate that it calls out specific lines because, of course line numbers change, especially in the almost 6 years since that ticket was created, but this PR doesn't seem connected to any TODO comments. Have you made an attempt to find what those TODO comment are / were and what the ticket actually asks to fix?

**Comment by kdaud** [comment]
> Have you made an attempt to find what those TODO comment are / were and what the ticket actually asks to fix?

I went through the file and got to find out the line numbers changed and based on the ticket description, I get to understand the the ticket requires using Spring's method injection to set the dependencies for the message service.

Am not so familiar with spring method injection but my initial work was to add a bean in the applicationContext-service.xml so that the Message service dependencies are set via lookup-method provided by the spring injection  mechanism. 

Could it be that the TODO is no-longer existing as of now? or  I needed to approach it differently?

**Comment by ibacher** [comment]
@kdaud They're still there. They've just moved to lines [668](https://github.com/kdaud/openmrs-core/blob/8db47aad4c28db1e7ff68c2f9733670fc03ec3ea/api/src/main/java/org/openmrs/api/context/Context.java#L668) and [678](https://github.com/kdaud/openmrs-core/blob/8db47aad4c28db1e7ff68c2f9733670fc03ec3ea/api/src/main/java/org/openmrs/api/context/Context.java#L678). The first comment is a bit unfortunate because I think it directed you to the wrong place.

With Spring, we already have a Spring context created and we should basically never create a new one.

**Pull request closed without merging**